### PR TITLE
📏 Completion length logging fix + remainder logging fix

### LINKED ISSUE #3476: [GRPO] completion lengths are incorrectly logged when mask_truncated_completions=True
### Reproduction

When the `mask_truncated_completions` is True, the values logged for the completion lengths are incorrect, for the max they are equal to the terminated completions and for in they are often zero. This is because we use the attn masks to calculate the lengths and these are zeroed for the truncated sequences.

The issue is obvious when you look at the min completion lengths:

Image

It should be a simple change to make a copy of the mask here:
https://github.com/huggingface/trl/blob/a528b9c465d618c4f2fda7bbcb145fd441e84224/trl/trainer/grpo_trainer.py#L1110

and then gathering the copied masks here:
https://github.com/huggingface/trl/blob/a528b9c465d618c4f2fda7bbcb145fd441e84224/trl/trainer/grpo_trainer.py#L1215

### System Info

- Platform: Linux-5.15.0-1049-aws-x86_64-with-glibc2.31
- Python version: 3.11.11
- TRL version: 0.18.0.dev0+a528b9c
- PyTorch version: 2.6.0
- accelerator(s): cpu
- Transformers version: 4.51.3
- Accelerate version: 1.3.0
- Accelerate config: not found
- Datasets version: 3.4.1
- HF Hub version: 0.30.2
- bitsandbytes version: 0.45.3
- DeepSpeed version: 0.16.4
- Diffusers version: 0.32.2
- Liger-Kernel version: 0.5.8
- LLM-Blender version: 0.0.2
- OpenAI version: 1.66.5
- PEFT version: 0.14.0
- vLLM version: 0.8.4

### Checklist

- [x] I have checked that my issue isn't already filed (see [open issues](https://github.com/huggingface/trl/issues?q=is%3Aissue))
- [x] I have included my system information
- [x] Any code provided is minimal, complete, and reproducible ([more on MREs](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any code provided is properly formatted in code blocks, (no screenshot, [more on code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks))
- [x] Any traceback provided is complete

### PR DESCRIPTION
**Pull request by shirinyamani**
# What does this PR do?

Fixes #3476

I'm testing using this tiny script on two gpus:

for server:
```python
CUDA_VISIBLE_DEVICES=0 trl vllm-serve --model Qwen/Qwen3-0.6B
```
trianing script: 

```python
from datasets import load_dataset
from trl import GRPOTrainer, GRPOConfig

dataset = load_dataset("trl-lib/tldr", split="train[:1%]")

# Dummy reward function: count the number of unique characters in the completions
def reward_num_unique_chars(completions, **kwargs):
    return [len(set(c)) for c in completions]

training_args = GRPOConfig(
    output_dir="qwen3-mask-completions",
    bf16=True,
    gradient_checkpointing=True,
    logging_steps=10,
    use_vllm=True,
    mask_truncated_completions=True,
)

trainer = GRPOTrainer(
    model="Qwen/Qwen3-0.6B",
    args=training_args,
    reward_funcs=reward_num_unique_chars,
    train_dataset=dataset,
)

trainer.train()
```
for training:
```python
CUDA_VISIBLE_DEVICES=0 accelerate launch trl/trainer/test.py
```

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/trl/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a GitHub issue? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by qgallouedec** [comment]
It seems like the PR isn't working, see the CI

**Comment by qgallouedec** [comment]
I'm taking over this PR so that we can have this fix in the next release.
